File: backend/app/models/category.py
from pydantic import BaseModel
from typing import Optional, List

class CategoryBase(BaseModel):
    name: str
    description: Optional[str] = None
    parent_id: Optional[str] = None  # For subcategories

class CategoryCreate(CategoryBase):
    pass

class Category(CategoryBase):
    id: str
    subcategories: List[str] = []  # List of subcategory IDs

class PresetCategory(BaseModel):
    name: str
    subcategories: List[str] = []

PRESET_CATEGORIES = [
    PresetCategory(
        name="Cold and Flu",
        subcategories=["Cough", "Fever", "Nasal Congestion"]
    ),
    PresetCategory(
        name="Pain Relief",
        subcategories=["Headache", "Muscle Pain", "Joint Pain"]
    ),
    PresetCategory(
        name="Allergies",
        subcategories=["Antihistamines", "Decongestants"]
    ),
    PresetCategory(
        name="Gastrointestinal",
        subcategories=["Antacids", "Anti-diarrheal", "Digestive Health"]
    ),
    PresetCategory(
        name="Heart and Circulation",
        subcategories=["Blood Pressure", "Heart Health"]
    ),
    PresetCategory(
        name="Vitamins and Supplements",
        subcategories=["Multivitamins", "Minerals", "Dietary Supplements"]
    )
] 

File: lib/models/category.dart
class Category {
  final String id;
  final String name;
  final String? description;
  final String? parentId;
  final List<String> subcategories;

  Category({
    required this.id,
    required this.name,
    this.description,
    this.parentId,
    this.subcategories = const [],
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      parentId: json['parent_id'],
      subcategories: (json['subcategories'] as List?)?.cast<String>() ?? [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'parent_id': parentId,
      'subcategories': subcategories,
    };
  }
} 